oidcclient/jwk.go

```diff
@@ -96,13 +102,14 @@ func (ks *OIDCTenant) updateJWKsMemory(zoneID string) (jwk.Set, error) {
 	return ks.jwks, nil
 }
 
-func (ks *OIDCTenant) getJWKsFromServer(zoneID string) (r interface{}, err error) {
+func (ks *OIDCTenant) getJWKsFromServer(appTID, clientID string) (r interface{}, err error) {
 	result := updateKeysResult{}
 	req, err := http.NewRequestWithContext(context.TODO(), http.MethodGet, ks.ProviderJSON.JWKsURL, http.NoBody)
 	if err != nil {
 		return result, fmt.Errorf("can't create request to fetch jwk: %v", err)
 	}
-	req.Header.Add(zoneIDHeader, zoneID)
+	req.Header.Add(appTIDHeader, appTID)
+	req.Header.Add(clientIDHeader, clientID)
```

Comment:
There seems to be an issue that if the headers are set both need to have an valid value. At least there was a fix for this in the java client lib: https://github.com/SAP/cloud-security-services-integration-library/pull/1272/files

Reply:
This has been a fix on client side because the IAS server could not handle if only one header was sent, but this is resolved by now and can also be removed in our client libs.